### PR TITLE
Adding automatic data disk performance meter template

### PR DESCRIPTION
**Pull request by AlekseiPolkovnikov**
### Contributing guide
https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md

### Changelog
*
*
*
*
*

### Description of the change

### CODE REVIEW

File: vm-disk-performance-meter/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+                          ===============  =========================  ==========
+                          testresult       String                     READ: io=2051.2MB, aggrb=78853KB/s, minb=19713KB/s, maxb=20024KB/s, mint=26222msec, maxt=26636msec; WRITE: io=2044.9MB, aggrb=78613KB/s, minb=19653KB/s, maxb=19963KB/s, mint=26222msec, maxt=26636msec;
+```
+
+![alt text](images/diskperformance.png "Disk performance measurement output")
+
+Subsequent performance measurements can have different results. To re-measure and get full ouput you can login to the test VM with credentials you provided during deployment and use this a command like this:
+
+```shell
+sudo echo -e '[io]\nrw=randrw\nsize=128m\ndirectory=/datadisk' | sudo fio -
```
Comment (bmoore-msft) [inline vm-disk-performance-meter/README.md]: is this the correct command "sudo fio -" - I was expecting something after the hyphen...

File: vm-disk-performance-meter/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+    "dataDiskHostCaching": {
+      "type": "string",
+      "defaultValue": "ReadWrite",
+      "allowedValues": [
+        "",
+        "ReadOnly",
+        "ReadWrite"
+      ],
+      "metadata": {
+        "description": "Storage Account type for the test disk"
```
Comment (bmoore-msft) [inline vm-disk-performance-meter/azuredeploy.json]: caching type

File: vm-disk-performance-meter/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+    "installCmd": "apt-get update > /dev/null && apt-get -y install fio > /dev/null",
+    "globalCmd": "[concat('echo [global] > t && echo size=', parameters('testSize'), ' >> t && echo direct=1 >> t && echo iodepth=256 >> t && echo ioengine=libaio >> t && echo bs=8k >> t')]",
+    "workerCmd": "[concat('echo rw=', parameters('testType'), ' >> t && echo directory=/datadisk >> t')]",
+    "cfgCmd": "[concat(variables('globalCmd'), ' && echo [w1] >> t && ', variables('workerCmd'), ' && echo [w2] >> t && ', variables('workerCmd'), ' && echo [w3] >> t && ', variables('workerCmd'), ' && echo [w4] >> t && ', variables('workerCmd'))]",
+    "warmupCmd": "[concat('fio --runtime ', parameters('secondsToRunTest'),  ' t > /dev/null')]",
+    "runCmd": "[concat('fio --runtime ', parameters('secondsToRunTest'),  ' t | grep -E \"READ:|WRITE:\" | tr \"\n\" \";\" | tr -s [:space:] | sed \"s/ :/:/g\" | sed \"s/= /=/g\" ')]",
+    "fullCmd": "[concat('sudo bash -c ', variables('qs'), variables('diskCmd'), ' && ', variables('installCmd'), ' && ', variables('cfgCmd'), ' && ', variables('runCmd'), variables('qs'))]"
```
Comment (bmoore-msft) [inline vm-disk-performance-meter/azuredeploy.json]: I don't think a human could decipher and/or modify this (though I'm quite impressed you wrote it :)) - can we break it out into an sh file so it's easier to digest and leverage?

File: vm-disk-performance-meter/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+    "nicName": "testVMNic",
+    "addressPrefix": "10.0.0.0/16",
+    "subnetName": "Subnet",
+    "subnetPrefix": "10.0.0.0/24",
+    "storageAccountType": "Standard_LRS",
+    "publicIPAddressName": "publicIP",
+    "publicIPAddressType": "Static",
+    "vmStorageAccountContainerName": "vhds",
+    "virtualNetworkName": "testVNET",
+    "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',variables('virtualNetworkName'))]",
```
Comment (bmoore-msft) [inline vm-disk-performance-meter/azuredeploy.json]: I don't think you need this var (see comment on subnetId)

File: vm-disk-performance-meter/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+    "scriptFolder": "scripts",
+    "scriptFileName": "testBandwidth.ps1",
+    "vhdStorageAccountName": "[concat(uniqueString(resourceGroup().id), 'storage')]",
+    "dataStorageAccountName": "[concat(uniqueString(resourceGroup().id), 'datastorage')]",
+    "frontEndNSGName": "[concat('webtestnsg-', uniqueString(resourceGroup().id))]",
+    "vmName": "testVM",
+    "subnetRef": "[concat(variables('vnetID'),'/subnets/',variables('subnetName'))]",
```
Comment (bmoore-msft) [inline vm-disk-performance-meter/azuredeploy.json]: can we use:
subnetref: resourceId('Microsoft.Network/virutualNetworks/subnets', variables('virtualNetworkName, variables('subnetName')

File: vm-disk-performance-meter/README.md (hunk 1 above)
Comment (AlekseiPolkovnikov) [inline vm-disk-performance-meter/README.md]: hyphen means: take stdin for fio, it is correct

File: vm-disk-performance-meter/azuredeploy.json (hunk 5 above)
Comment (AlekseiPolkovnikov) [inline vm-disk-performance-meter/azuredeploy.json]: thanks, fixed

File: vm-disk-performance-meter/azuredeploy.json (hunk 4 above)
Comment (AlekseiPolkovnikov) [inline vm-disk-performance-meter/azuredeploy.json]: thanks, fixed

File: vm-disk-performance-meter/azuredeploy.json (hunk 2 above)
Comment (AlekseiPolkovnikov) [inline vm-disk-performance-meter/azuredeploy.json]: thanks, fixed

**Comment by AlekseiPolkovnikov** [comment]
@bmoore-msft thanks, done

**Comment by AlekseiPolkovnikov** [comment]
Re-opening to re-run the checks

**Pull request merged**